chain/message_store.go

```diff
@@ -41,7 +42,7 @@ func (ms *MessageStore) LoadMessages(ctx context.Context, c cid.Cid) ([]*types.S
 	// TODO #1324 message collection shouldn't be a slice
 	var out []*types.SignedMessage
 	err := ms.ipldStore.Get(ctx, c, &out)
-	return out, err
+	return out, errors.Wrapf(err, "could not load message collection with cid: %s", c.String())
```

Comment:
This seems to give the impression all of the messages go into a piece of hashable content. I was under the impression messages would be broken up over multiple hashes. Did I misunderstand? No big worry either way. It makes the changes to Graphsync easier.

Reply:
You've got it.  If I understand correctly we want a single hash to reference the message collection from the block header to keep header size low.  We eventually are planning to do some [merkelized data structure](https://github.com/filecoin-project/specs/blob/dd3bc99c76b2b6bd0baeb23687b0fe5e01c2db57/sharray.md) which will require graphsync traversals across several links, but we'll always have a single hash in the header and that merkelized data structure is further out.